modules/layers/src/path-layer/path-tesselator.js

```diff
@@ -0,0 +1,182 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+import Tesselator from './tesselator';
+import {fp64 as fp64Module} from 'luma.gl';
+const {fp64LowPart} = fp64Module;
+
+// This class is set up to allow querying one attribute at a time
+// the way the AttributeManager expects it
+export default class PathTesselator extends Tesselator {
+  constructor({data, getGeometry, fp64}) {
+    super({
+      data,
+      getGeometry,
+      fp64,
+      attributes: {
+        startPositions: {size: 3},
+        endPositions: {size: 3},
+        leftDeltas: {size: 3},
+        rightDeltas: {size: 3},
+        startEndPositions64XyLow: {size: 4, fp64: true}
+      }
+    });
+  }
+
+  /* Getters */
+  startPositions() {
+    return this.attributes.startPositions;
+  }
+
+  endPositions() {
+    return this.attributes.endPositions;
+  }
+
+  startEndPositions64XyLow() {
+    return this.attributes.startEndPositions64XyLow;
+  }
+
+  leftDeltas() {
+    return this.attributes.leftDeltas;
+  }
+
+  rightDeltas() {
+    return this.attributes.rightDeltas;
+  }
+
+  strokeWidths({target, getWidth}) {
+    return this._updateAttribute({
+      target,
+      size: 1,
+      getValue: object => [getWidth(object)]
+    });
+  }
+
+  dashArrays({target, getDashArray}) {
+    return this._updateAttribute({target, size: 2, getValue: getDashArray});
+  }
+
+  colors({target, getColor}) {
+    return this._updateAttribute({
+      target,
+      size: 4,
+      getValue: object => {
+        const color = getColor(object);
+        if (isNaN(color[3])) {
+          color[3] = 255;
+        }
+        return color;
+      }
+    });
+  }
+
+  pickingColors({target, getPickingColor}) {
+    return this._updateAttribute({
+      target,
+      size: 3,
+      getValue: (object, index) => getPickingColor(index)
+    });
+  }
+
+  /* Utilities */
+  getPathLength(path) {
+    return path.length;
+  }
+
+  getPointOnPath(path, index) {
+    return path[index];
+  }
+
+  countInstancesInGeometry(path) {
+    return Math.max(0, this.getPathLength(path) - 1);
+  }
+
+  isClosed(path) {
+    const numPoints = this.getPathLength(path);
+    const firstPoint = this.getPointOnPath(path, 0);
+    const lastPoint = this.getPointOnPath(path, numPoints - 1);
+    return (
+      firstPoint[0] === lastPoint[0] &&
+      firstPoint[1] === lastPoint[1] &&
+      firstPoint[2] === lastPoint[2]
+    );
+  }
+
+  /* Updaters */
+  /* eslint-disable max-statements, complexity */
+  updateGeometryAttributes() {
```

Comment:
I expect that updating one path at time is for a separate PR. The base Tesselator would call this function once for each path.